contrib/win32/win32compat/win32_monitor_wrap.c

```diff
@@ -211,4 +211,44 @@ int mm_load_profile(const char* user_name, u_int token)
 	}
 
 	return result;
+}
+
+void*
+mm_auth_custom_lsa(const char* user, const char* password, const char* domain, const char* lsa_auth_pkg)
+{
+	/* Pass credentials to privileged agent to retrieve token upon successful authentication */
+	struct sshbuf *msg = NULL;
+	HANDLE token = 0;
+	int agent_fd;
+
+	while (1) {
+		if ((agent_fd = get_priv_agent_sock()) == -1)
+			break;
+
+		msg = sshbuf_new();
+		if (!msg)
+			fatal("%s: out of memory", __func__);
+
+		if (sshbuf_put_u8(msg, SSH_PRIV_AGENT_MSG_ID) != 0 ||
+			sshbuf_put_cstring(msg, CUSTOM_LSA_AUTH_REQUEST) != 0 ||
+			sshbuf_put_cstring(msg, user) != 0 ||
+			sshbuf_put_cstring(msg, password) != 0 ||
+			sshbuf_put_cstring(msg, domain) != 0 ||
+			sshbuf_put_cstring(msg, lsa_auth_pkg) != 0 ||
+			ssh_request_reply(agent_fd, msg, msg) != 0) {
```

Comment:
for troubleshooting purpose, can we separate the sh_request_reply from the sending and log debug info?

Reply:
fixed

Reply:
ssh_request_reply is still in the same if condition